### ISSUE TITLE
NoMethodError: undefined method `size' for #<Rack::Lint::InputWrapper>

### ISSUE THREAD

**Issue by sthakur** [adamlogic/rails_autoscale_agent#1]
Autoscale works on Heroku in production envirnoment but with unicorn (4.8.3) and foreman (0.76.0) in my development environment, I get the following error:

11:49:29 web.1    | NoMethodError: undefined method `size' for #<Rack::Lint::InputWrapper:0x007fb3c4763e30>
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/rails_autoscale_agent-0.3.0/lib/rails_autoscale_agent/request.rb:10:in `initialize'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/rails_autoscale_agent-0.3.0/lib/rails_autoscale_agent/middleware.rb:19:in `new'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/rails_autoscale_agent-0.3.0/lib/rails_autoscale_agent/middleware.rb:19:in `block in call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/activesupport-4.1.0/lib/active_support/tagged_logging.rb:68:in `block in tagged'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/activesupport-4.1.0/lib/active_support/tagged_logging.rb:26:in `tagged'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/activesupport-4.1.0/lib/active_support/tagged_logging.rb:68:in `tagged'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/rails_autoscale_agent-0.3.0/lib/rails_autoscale_agent/middleware.rb:17:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/newrelic_rpm-3.9.0.229/lib/new_relic/agent/instrumentation/middleware_tracing.rb:57:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/activesupport-4.1.0/lib/active_support/cache/strategy/local_cache_middleware.rb:26:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/newrelic_rpm-3.9.0.229/lib/new_relic/agent/instrumentation/middleware_tracing.rb:57:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/rack-1.5.2/lib/rack/lock.rb:17:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/newrelic_rpm-3.9.0.229/lib/new_relic/agent/instrumentation/middleware_tracing.rb:57:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/actionpack-4.1.0/lib/action_dispatch/middleware/static.rb:64:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/newrelic_rpm-3.9.0.229/lib/new_relic/agent/instrumentation/middleware_tracing.rb:57:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/rack-1.5.2/lib/rack/sendfile.rb:112:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/newrelic_rpm-3.9.0.229/lib/new_relic/agent/instrumentation/middleware_tracing.rb:57:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/rack-cors-0.4.0/lib/rack/cors.rb:80:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/newrelic_rpm-3.9.0.229/lib/new_relic/agent/instrumentation/middleware_tracing.rb:57:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/sentry-raven-0.9.4/lib/raven/rack.rb:59:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/newrelic_rpm-3.9.0.229/lib/new_relic/agent/instrumentation/middleware_tracing.rb:57:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/rack-mini-profiler-0.9.2/lib/mini_profiler/profiler.rb:215:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/newrelic_rpm-3.9.0.229/lib/new_relic/agent/instrumentation/middleware_tracing.rb:57:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/railties-4.1.0/lib/rails/engine.rb:514:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/railties-4.1.0/lib/rails/application.rb:144:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/newrelic_rpm-3.9.0.229/lib/new_relic/agent/instrumentation/middleware_tracing.rb:57:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/rack-1.5.2/lib/rack/lint.rb:49:in `_call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/rack-1.5.2/lib/rack/lint.rb:37:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/newrelic_rpm-3.9.0.229/lib/new_relic/agent/instrumentation/middleware_tracing.rb:57:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/rack-1.5.2/lib/rack/showexceptions.rb:24:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/newrelic_rpm-3.9.0.229/lib/new_relic/agent/instrumentation/middleware_tracing.rb:57:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/rack-1.5.2/lib/rack/commonlogger.rb:33:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/sinatra-1.4.6/lib/sinatra/base.rb:218:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/newrelic_rpm-3.9.0.229/lib/new_relic/agent/instrumentation/middleware_tracing.rb:57:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/rack-1.5.2/lib/rack/chunked.rb:43:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/newrelic_rpm-3.9.0.229/lib/new_relic/agent/instrumentation/middleware_tracing.rb:57:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/rack-1.5.2/lib/rack/content_length.rb:14:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/newrelic_rpm-3.9.0.229/lib/new_relic/agent/instrumentation/middleware_tracing.rb:57:in `call'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/unicorn-4.8.3/lib/unicorn/http_server.rb:576:in `process_client'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/unicorn-4.8.3/lib/unicorn/http_server.rb:670:in `worker_loop'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/unicorn-4.8.3/lib/unicorn/http_server.rb:525:in `spawn_missing_workers'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/unicorn-4.8.3/lib/unicorn/http_server.rb:140:in `start'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/gems/unicorn-4.8.3/bin/unicorn:126:in `<top (required)>'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/bin/unicorn:23:in `load'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/bin/unicorn:23:in `<main>'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/bin/ruby_executable_hooks:15:in `eval'
11:49:29 web.1    | 	/Users/johndoe/.rvm/gems/ruby-2.1.5/bin/ruby_executable_hooks:15:in `<main>'

**Comment by adamlogic** [comment]
Thanks for reporting this. I just pushed a fix up to a branch. If you could verify that it works for you, I'll merge it in and release a new gem.

```ruby
gem 'rails_autoscale_agent', github: 'adamlogic/rails_autoscale_agent', branch: 'fix-rack-input-size'
```

**Comment by sthakur** [comment]
Perfect! Thanks for taking care of this so promptly.

**Comment by adamlogic** [comment]
No problem! Fixed in 5eacc26b7fcc0f7c5c3420e6080a8781bb71f0ea.
